exoscale_auth.py

```diff
@@ -53,7 +53,7 @@ def _sign_request(self, request, expiration_ts):
         ))
 
         # Request body
-        msg_parts.append(body.encode('utf-8') if request.body else u'')
+        msg_parts.append(str(request.body or b'', 'utf-8') if hasattr(request, 'body') else u'')
```

Comment:
```suggestion
        if hasattr(request, "body"):
            body = request.body.decode("utf-8")
        else:
            body = ""
        msg_parts.append(body)
```

Reply:
Slightly adjusted variant in 2c3a121920d5ec92dfdc6d52f198a415f17d5edc.